crates/router/src/utils/user.rs

```diff
@@ -207,16 +207,62 @@ pub fn get_redis_connection(state: &SessionState) -> UserResult<Arc<RedisConnect
         .attach_printable("Failed to get redis connection")
 }
 
-impl ForeignFrom<user_api::AuthConfig> for UserAuthType {
-    fn foreign_from(from: user_api::AuthConfig) -> Self {
+impl ForeignFrom<&user_api::AuthConfig> for UserAuthType {
+    fn foreign_from(from: &user_api::AuthConfig) -> Self {
         match from {
-            user_api::AuthConfig::OpenIdConnect { .. } => Self::OpenIdConnect,
-            user_api::AuthConfig::Password => Self::Password,
-            user_api::AuthConfig::MagicLink => Self::MagicLink,
+            &user_api::AuthConfig::OpenIdConnect { .. } => Self::OpenIdConnect,
+            &user_api::AuthConfig::Password => Self::Password,
+            &user_api::AuthConfig::MagicLink => Self::MagicLink,
         }
     }
 }
 
+pub async fn construct_public_and_private_db_configs(
+    auth_config: &user_api::AuthConfig,
+    encryption_key: &[u8],
+) -> UserResult<(Option<Encryption>, Option<serde_json::Value>)> {
```

Comment:
why `(Option<T1>,Option<T2>)` why not `Option<(T1,T2)>` if public and private config are coupled ?

Reply:
In DB, both are optional, so a case where when one is optional and other is not comes then, we will have to change this everywhere.